adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -2246,6 +2287,49 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
     }
   }
 
+  /**
+   * Load a path name in Parquet + Avro format into a VariantContextRDD.
+   *
+   * @param pathName The path name to load genotypes from.
+   *   Globs/directories are supported.
+   * @return Returns a GenotypeRDD.
+   */
+  def loadVariantContexts(
+    pathName: String): VariantContextRDD = {
+
+    if (isVcfExt(pathName)) {
+      loadVcf(pathName)
+    } else {
+      loadParquetVariantContexts(pathName)
+    }
+  }
+
+  /**
+   * Load a path name in Parquet + Avro format into a VariantContextRDD.
+   *
+   * @param pathName The path name to load genotypes from.
+   *   Globs/directories are supported.
+   * @return Returns a GenotypeRDD.
+   */
+  def loadParquetVariantContexts(
+    pathName: String): VariantContextRDD = {
+
+    // load header lines
+    val headers = loadHeaderLines(pathName)
+
+    // load sequence info
+    val sd = loadAvroSequenceDictionary(pathName)
+
+    // load avro record group dictionary and convert to samples
+    val samples = loadAvroSamples(pathName)
+
+    val sqlContext = SQLContext.getOrCreate(sc)
+    import sqlContext.implicits._
+    val ds = sqlContext.read.parquet(pathName).as[VariantContextProduct]
+
+    new DatasetBoundVariantContextRDD(ds, sd, samples, headers)
```

Comment:
we're intentionally bound to dataset here, is the idea to move other loadParquetXxx methods in the same direction?

Reply:
Nah, we don't have an Avro impl of VariantContext, so we can't use the normal loadParquet method.

Reply:
ok